### DISCUSSION TITLE
How to detect confirm email if click or not?

### DISCUSSION THREAD

**Discussion by donPuerto** [supabase/supabase#6740] [Questions] [answered]
It is possible to detect the auth users response if the confirm email is click or now?

**Comment by GaryAustin1** [comment, marked as answer]
You probably are going to need to provide more info, but the link, if set up correctly, should launch your app and then the Supabase client will look at the token in the url to launch signin.  You should see activity on onAuthStateChange().

**Reply by donPuerto** [reply]
Hi @GaryAustin1.

Why when we do Signup the role in auth user is authenticated? see below
![image](https://user-images.githubusercontent.com/9329005/167731702-6512290f-017c-4f66-b32b-71cf6ca221c1.png)

There should be a flag whether to check  the confirm email link was click or not by the user
![image](https://user-images.githubusercontent.com/9329005/167732573-de5f8315-9bf4-4ec1-b8aa-ed4d6eb22877.png)

Scenario #1
If I already Signup, but email confirmation was not click, and when I register again, I have the ability the to redirect it to a certain page that saying you need click the confirm email first.

**Reply by AliQ80** [reply]
I'll post my solution here maybe it will help someone,

on the user object, I check for "confirmation_sent" , if true it means confirmation sent but not clicked yet
otherwise I check for "confirmed_at" if true means confirmation link clicked

not sure if this is the best solution, but this is my understanding, hope it helps

**Reply by christy23francis** [reply]
@AliQ80 How you did the confirmed_at check? Do you created a new trigger for that?

**Reply by AliQ80** [reply]
> @AliQ80 How you did the confirmed_at check? Do you created a new trigger for that?
> 

@christy23francis I have a function in pinia store that looks authentic user object and triggers a boolean,  so if there is a value I assume the email is confirmed, otherwise if undefined then it's not confirmed,  I hope this helps. 

```js
    async emailLogin(value: { email: string; password: string }) { 
       const client = useSupabaseClient() 
       try { 
         const { 
           data: { user }, 
           error, 
         } = await client.auth.signInWithPassword({ 
           email: value.email, 
           password: value.password, 
         }) 
         if (user) { 
           const { data } = await client 
             .from('profiles') 
             .select('username,id') 
             .eq('id', user.id) 
             .single() 
  
           this.uid = data.id 
           this.name = data.username 
           this.email = user.email 
           this.error = '' 

           if (user.confirmed_at) { 
             this.confirmed = true 
           } 

         } 
         if (error) throw error 
       } catch (error) { 
         this.name = '' 
         this.email = '' 
         this.error = error.message 
         this.confirmed = false 
       } 
     },
```
